lib/salus/npm_audit_wrapper.rb

```diff
@@ -0,0 +1,239 @@
+              radv.fetch(:module_name),
+              radv.fetch(:title),
+              radv.fetch(:severity),
+              radv.fetch(:url),
+              prod,
+              excepted
+            )
+          end
+
+          # Sort advisories with un-excepted prod vulnerabilities first,
+          # and in ascending order of ID otherwise
+          advisories.sort_by! do |advisory|
+            unexcepted_and_prod = !advisory.excepted? && advisory.prod?
+            [(unexcepted_and_prod ? 0 : 1), advisory.id.to_i]
+          end
+
+          advisories_by_id = advisories.map(&:id).zip(advisories).to_h
+
+          # The build fails iff there exist un-expect advisories against prod deps
+          failing_advisory_ids = advisories.select { |adv| !adv.excepted? && adv.prod? }.map(&:id)
+
+          # For informational purposes, we record any exceptions that don't refer
+          # to an actual advisory
+          extraneous_exceptions = @exceptions.reject { |id| advisories_by_id.key?(id) }
+
+          # Also, we record any exceptions that apply only to dev dependencies
+          extraneous_dev_exceptions = @exceptions.select do |id|
+            adv = advisories_by_id[id]
+            !adv.nil? && !adv.prod?
+          end
+
+          if advisories.any?
+            log(tabulate_advisories(advisories))
+          else
+            log(colorize('There are no advisories against your dependencies. Hooray!', :green))
+          end
+
+          if failing_advisory_ids.any?
+            s = failing_advisory_ids.join(', ')
+            log("Audit failed pending the following advisory(s): #{s}.")
+            log('To fix the build, please resolve the previous advisory(s), or add exceptions.')
+          end
+
+          if extraneous_exceptions.any? || extraneous_dev_exceptions.any?
+            if extraneous_exceptions.any?
+              log(
+                'The following exception(s) do not match any advisory against the directory: ' \
+                "#{extraneous_exceptions.join(', ')}."
+              )
+            end
+
+            if extraneous_dev_exceptions.any?
+              log(
+                'The following exceptions apply only to development dependencies: ' \
+                "#{extraneous_dev_exceptions.join(', ')}."
+              )
+            end
+
+            log('These exceptions can safely be removed.')
+          end
+
+          failing_advisory_ids.empty?
+        end
+      end
+    end
+
+    private
+
+    def log(string)
+      @stream.puts(string)
+    end
+
+    def ensure_package_lock!
+      created_package_lock = false
+
+      if !File.exist?('package-lock.json')
+        log('Creating a temporary package-lock.json...')
+        run_command!('npm install --package-lock-only', raise_on_failure: true)
+        created_package_lock = true
+      end
+
+      yield
+    ensure
+      if created_package_lock
+        File.delete("package-lock.json")
+        log('Removed temporary package-lock.json')
```

Comment:
this should only be logged if verbose option (which can be given to Salus via a `-v` flag).
Might want to do this in a separate PR to keep this one small.